internal/base_component.go

```diff
@@ -111,6 +112,14 @@ func (base *BaseComponent) IsDirEmpty(options IsDirEmptyOptions) bool {
 	return false
 }
 
+func (base *BaseComponent) DeleteEmptyDirs(options DeleteDirOptions) error {
+	if base.next != nil {
+		return base.next.DeleteEmptyDirs(options)
+	}
+	// this method is only implemented in file cache
+	return fmt.Errorf("%s component does not implement DeleteEmptyDirs", base.Name())
```

Comment:
we shall return nil here, not all components will implement this method.

Reply:
`DeleteEmptyDirs` method is only implemented in file cache and is only called in `libfuse_rmdir` method when directory is not empty.
```
empty := fuseFS.NextComponent().IsDirEmpty(internal.IsDirEmptyOptions{Name: name})
if !empty {
// delete empty directories from local cache directory
err := fuseFS.NextComponent().DeleteEmptyDirs(internal.DeleteDirOptions{Name: name})
if err != nil {
	log.Err("Libfuse::libfuse_rmdir : Failed to delete %s [%s]", name, err.Error())
	return -C.ENOTEMPTY
    }
}
```

If file cache is not present in the config, this should ideally be pass through method. So, if nil is returned in base component, it won't return `C.ENOTEMPTY` error.

Reply:
If there are all components in the pipeline and tommorow for some reason we decide to forward the call, it will reach azstorage and it will return error if it has not implemented this method. For all other methods as well we just return back nil if they  have not implemented it. Just to be consistent with other methods also we shall return nil and not error.

Reply:
Updated the return type of this method to return boolean value and error. For components which do not implement this, the default value returned is false and nil error.